scripts/combined-haddock.sh

```diff
@@ -0,0 +1,115 @@
+# This script generates a local, self-contained, deployable Haddock for the
+# Plutus project. It uses the experimental 'haddock-project' Cabal command,
+# which currently has some issues that need fixing. 
+
+# Setup source directory 
+SRC=combined_haddock_src
+rm -rf $SRC
+mkdir -p $SRC
+
+# Write haddock prologue 
+cat << EOF > $SRC/haddock.prologue
+= Combined Plutus Documentation
+    
+* "PlutusTx": Compiling Haskell to PLC (Plutus Core; on-chain code).
+* "PlutusTx.Prelude": Haskell prelude replacement compatible with PLC.
+* "PlutusCore": Programming language in which scripts on the Cardano blockchain are written.
+* "UntypedPlutusCore": On-chain Plutus code.
+EOF
+
+# Clean project and build haddock
+cabal clean 
+cabal update
+cabal build all
+cabal haddock-project \
+  --quickjump \
+  --gen-contents \
+  --hyperlinked-source \
+  --gen-index \
+  --internal \
+  --output=$SRC \
+  --prologue=$SRC/haddock.prologue
+
+# Setup destination directory 
+DST=combined_haddock_dst
+rm -rf $DST
+mkdir -p $DST
+
+# List of target haskell packages 
+# TODO use `find` to avoid hard-coding the list
+PACKAGE_NAMES=(
```

Comment:
I don't think this list is complete - e.g., it doesn't have `containers`.

It would be nice if we didn't have to build haddock for `base` and other third party libraries, but link to their haddock on Hackage. What would it take to do so?

Reply:
We are not exactly building haddock for base and other third party libraries, we are just making all hyperlinks point to our local copy. 
Unfortunately `cabal project-haddock` still has some limitations and this is as far as we can go. 
We can wait until better support is available in cabal.

As for the incomplete list -- I update the script